### ISSUE TITLE
Elevation info needs a better icon

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#14744]
### Describe your problem!

I talked to some beta users and even myself was fooled by the elevation info on the map.
As it shows "a ramp" users might first thing, that it shows an elevation difference (e.g. if a route is active in parallel).
This is emphasized by the fact, that its shown in the same screen area, which normally only shows distances to target, routing length, etc.

At least we should IMHO try to find a better icon, ideally we could also find a better place to show that info on the map screen (e.g. upper left instead upper right) to visualize its not related to routing and distance information.

Ideas for icons....did not check usage conditions:
![image](https://github.com/cgeo/cgeo/assets/949669/f8c68fcc-1518-4487-b746-67956d2ed281)
Source: https://www.flaticon.com/de/kostenloses-icon/elevation_9620066

### How to reproduce?

see above

### Reproducible

Yes

### c:geo Version

Nightly/Beta

**Comment by Lineflyer** [comment]
While discussing with a fellow geocacher just now:
He agreed, that the position and icon is not very clear. He suggested to show the info next to the position arrow of my position on the map and forget about an icon.

**Comment by moving-bits** [comment]
FYI: Currently we use characters (Unicode items), not icons in the distance & elevation info boxes.
If we want to use icons, we would need to redesign that. I remember having tried that back then, but for some reason that did not work visually (probably background or some other issues. do not remember right now).

**Comment by moving-bits** [comment]
> He suggested to show the info next to the position arrow of my position on the map

wouldn't that potentially hide things right in the spot where you are mostly interested in?

**Comment by Lineflyer** [comment]
That's a challenge, here is how it shows in Locus:
![image](https://github.com/cgeo/cgeo/assets/949669/47c1e00f-c319-40f0-961f-41f3a97c9dc4)

**Comment by eddiemuc** [comment]
> FYI: Currently we use characters (Unicode items), not icons in the distance & elevation info boxes.
> If we want to use icons, we would need to redesign that. I remember having tried that back then, but for some reason that did not work visually (probably background or some other issues. do not remember right now).

@moving-bits have you tried using a TextParam? It can take an icon and display It as part of a simple TextView. Something like the following (this is out of my head, so methid names/usage may differ slightly):


```
TextView tv = ... //TextView tonfill
int iconId = ... // resource I'd of icon to use
String userText = ... // text to display

final imageParam = ImageParam.id(iconId);
TextParam.text(userText).withImage(imageParam).applyTo(tv);
```

**Comment by eddiemuc** [comment]
This ia used e.g. to display icons next to the menu entries for context menus in Cache log tab:

![Screenshot_2023-10-20-07-29-57-60_a916ceb0c8a82619faeb76fe6b8ae997.jpg](https://github.com/cgeo/cgeo/assets/6909759/b5fd0186-4bb6-42c4-9ee8-8828a8679cac)

**Comment by moving-bits** [comment]
> @moving-bits have you tried using a TextParam?

Thanks for the pointer. I'm happy to give it a try in case we decide to stay with top-right position.

> > > He suggested to show the info next to the position arrow of my position on the map
> > 
> > wouldn't that potentially hide things right in the spot where you are mostly interested in?
> >
> That's a challenge, here is how it shows in Locus:

I understand the reasoning, but I'm not sure if I like this solution - it still clutters the map right around the position where I'm currently most interested in, IMHO. Elevation info normally isn't a top crucial info, so it's ok, IMHO, if it's displayed out of main focus (but still at hand without having to manually activate it for every reading).

If we really don't want to have it close to the other numbers in top right position, an alternative approach could be to put it as default info at the bottom of the long press popup menu. Then it's still easier to reach than some menu entry but does not block any map space at all normally.

But maybe using a more self-explanatory icon will already solve the issue and we can keep top-right position?

**Comment by moving-bits** [comment]
Material symbols (Google Fonts) gives you this icon:

![image](https://github.com/cgeo/cgeo/assets/3754370/baf80169-96cc-4199-a299-d13707ae367c)

(https://fonts.google.com/icons?icon.query=elevation)

**Comment by moving-bits** [comment]
PR #14752 shows a proposal with new icons for both elevation info and routing distance info

**Comment by Lineflyer** [comment]
> If we really don't want to have it close to the other numbers in top right position, an alternative approach could be to put it as default info at the bottom of the long press popup menu.

Its an option, the user actively needs to turn on, isnt it?
Then I dont see a problem showing it next to the position icon then. The user is interested in it and he can also easily move the map if something is covered below. 

The top right is IMHO not at all the right position for an information not related to a route. 

Shall we take this to a new issue?

**Comment by Lineflyer** [comment]
Let me use this closed topic to get some more feedback:

First of all the improved icons are really great, but the map elevation info UI is still a bit misleading.
I showed the following screen to 8 different people and all of them interpreted the elevation info as "This is the elevation difference to go on my route".

In fact that would be a valuable information at this place (that would be a different feature request) but I still think its not the right place and not the right icon to present the current location elevation on the map.

![image](https://github.com/cgeo/cgeo/assets/949669/df1c23f7-27b0-4cba-a3c0-74b2fc62d3c0)

**Comment by fm-sys** [comment]
> In fact that would be a valuable information at this place (that would be a different feature request)

+1 for (optionally) showing route height meters. Although the question arises, should it be the height meters to the current target cache, individual route, etc?

**Comment by ztNFny** [comment]
As long as it's placed right below route info items it'll alway be associated with routing imho.
How about showing it in another place - and maybe together with the users current coordinates? Altitude is really just 1 of the 3 dimensionsy if they are shown together itll be clear.
